### PR TITLE
Add KIE Server basic S2I template and adjust test scenarios to run S2…

### PR DESCRIPTION
**Pull request by jakubschwan**
… templates

### CODE REVIEW

File: framework-cloud/framework-openshift/src/main/java/org/kie/cloud/openshift/settings/builder/KieServerBasicS2ISettingsBuilderImpl.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+        envVariables.put(OpenShiftTemplateConstants.KIE_SERVER_PWD, DeploymentConstants.getKieServerPassword());
+    }
+
+    @Override
+    public DeploymentSettings build() {
+        return new DeploymentSettingsImpl(envVariables, appTemplate);
+    }
+
+    @Override
+    public KieServerS2ISettingsBuilder withSecuredHostame(String https) {
```
Comment (sutaakar) [inline framework-cloud/framework-openshift/src/main/java/org/kie/cloud/openshift/settings/builder/KieServerBasicS2ISettingsBuilderImpl.java]: I am not sure if this is the best solution for this case.
We should either throw an exception as user is trying to set parameter which cannot be configured or create a specific interface which wouldn't contain this method.

File: test-cloud/test-cloud-remote/src/test/java/org/kie/cloud/integrationtests/architecture/ManagedKieServersWithSmartRouterAndControllerIntegrationTest.java [hunk 2]
```diff
@@ -118,7 +148,7 @@ protected GenericScenario createDeploymentScenario(DeploymentScenarioBuilderFact
     }
 
     private DeploymentSettings configureKieServerS2I(DeploymentScenarioBuilderFactory deploymentScenarioFactory, String applicationName, String containerDeploymnet) {
-        return deploymentScenarioFactory.getKieServerS2ISettingsBuilder()
+        return createDeploymentSettings()
```
Comment (sutaakar) [inline test-cloud/test-cloud-remote/src/test/java/org/kie/cloud/integrationtests/architecture/ManagedKieServersWithSmartRouterAndControllerIntegrationTest.java]: You can directly reference parameterized parameter here, no need for a method.

File: test-cloud/test-cloud-remote/src/test/java/org/kie/cloud/integrationtests/architecture/UnmanagedKieServersWithSmartRouterAndControllerIntegrationTest.java [hunk 3]
```diff
@@ -103,8 +132,8 @@ protected GenericScenario createDeploymentScenario(DeploymentScenarioBuilderFact
                 .build();
     }
 
-    private DeploymentSettings configureKieServerS2I(DeploymentScenarioBuilderFactory deploymentScenarioFactory, String applicationName, String containerDeploymnet) {
-        return deploymentScenarioFactory.getKieServerS2ISettingsBuilder()
+    private DeploymentSettings configureKieServerS2I(String applicationName, String containerDeploymnet) {
+        return createDeploymentSettings()
```
Comment (sutaakar) [inline test-cloud/test-cloud-remote/src/test/java/org/kie/cloud/integrationtests/architecture/UnmanagedKieServersWithSmartRouterAndControllerIntegrationTest.java]: same here

File: test-cloud/test-cloud-remote/src/test/java/org/kie/cloud/integrationtests/architecture/UnmanagedKieServersWithSmartRouterIntegrationTest.java [hunk 4]
```diff
@@ -83,18 +111,18 @@ protected GenericScenario createDeploymentScenario(DeploymentScenarioBuilderFact
...
         return deploymentScenarioFactory.getGenericScenarioBuilder()
                 .withSmartRouter(smartRouter)
                 .withKieServer(kieServerABC, kieServerDEF, kieServerGHI)
                 .build();
     }
 
-    private DeploymentSettings configureKieServerS2I(DeploymentScenarioBuilderFactory deploymentScenarioFactory, String applicationName, String containerDeploymnet) {
-        return deploymentScenarioFactory.getKieServerS2ISettingsBuilder()
+    private DeploymentSettings configureKieServerS2I(String applicationName, String containerDeploymnet) {
+        return createDeploymentSettings()
```
Comment (sutaakar) [inline test-cloud/test-cloud-remote/src/test/java/org/kie/cloud/integrationtests/architecture/UnmanagedKieServersWithSmartRouterIntegrationTest.java]: and here

File: test-cloud/test-cloud-remote/src/test/java/org/kie/cloud/integrationtests/s2i/KieServerS2iDroolsIntegrationTest.java [hunk 5]
```diff
@@ -72,7 +101,7 @@
     protected GenericScenario createDeploymentScenario(DeploymentScenarioBuilderFactory deploymentScenarioFactory) {
         repositoryName = gitProvider.createGitRepositoryWithPrefix("KieServerS2iDroolsRepository", ClassLoader.class.getResource(PROJECT_SOURCE_FOLDER).getFile());
 
-        DeploymentSettings kieServerS2Isettings = deploymentScenarioFactory.getKieServerS2ISettingsBuilder()
+        DeploymentSettings kieServerS2Isettings = createDeploymentSettings()
```
Comment (sutaakar) [inline test-cloud/test-cloud-remote/src/test/java/org/kie/cloud/integrationtests/s2i/KieServerS2iDroolsIntegrationTest.java]: and here

File: framework-cloud/framework-openshift/src/main/java/org/kie/cloud/openshift/settings/builder/KieServerBasicS2ISettingsBuilderImpl.java (hunk 1 above)
Comment (jakubschwan) [reply framework-cloud/framework-openshift/src/main/java/org/kie/cloud/openshift/settings/builder/KieServerBasicS2ISettingsBuilderImpl.java]: Added Runtime exception.

**Pull request merged**